lib/jxl/encode.cc

```diff
@@ -884,24 +902,24 @@ JxlEncoderStatus JxlEncoderAddImageFrame(const JxlEncoderOptions* options,
       c_current = jxl::ColorEncoding::SRGB(pixel_format->num_channels < 3);
     }
   } else {
-    c_current = options->enc->metadata.m.color_encoding;
+    c_current = frame_options->enc->metadata.m.color_encoding;
   }
 
-  size_t xsize = options->enc->metadata.xsize();
-  size_t ysize = options->enc->metadata.ysize();
-  if (options->values.cparams.already_downsampled) {
-    size_t factor = options->values.cparams.resampling;
+  size_t xsize = frame_options->enc->metadata.xsize();
+  size_t ysize = frame_options->enc->metadata.ysize();
```

Comment:
Here we would then use the frame dimensions, not the image dimensions, and also set the frame header fields (offset, duration, blend mode etc), right?

Reply:
this change in encode.cc was unintended as it's a header-only change, this comment is now obsolete